ha/core/event_analyzer/parser/parser.py

```diff
@@ -159,15 +159,102 @@ class ClusterResourceParser(Parser):
+                key = self.system_health_node_key_template.replace(pattern, config_val)
+            self.system_health_node_key_template = key
+        return self.system_health_node_key_template
+
+    def _create_health_event_format(self, cluster_resource_alert: dict, specific_info: dict = None) -> dict:
+        """
+        Creates the defined structure to send an helath event to system health
+        for further processing
+        """
+        timestamp = str(int(time.time()))
+        event_id = timestamp + str(uuid.uuid4().hex)
+        node_id = cluster_resource_alert["_resource_name"]
+        resource_type = cluster_resource_alert["_resource_type"]
+        event_type = cluster_resource_alert["_event_type"]
+        timestamp = cluster_resource_alert["_timestamp"]
+
+        event = {
+                    EVENT_ATTRIBUTES.EVENT_ID : event_id,
+                    EVENT_ATTRIBUTES.EVENT_TYPE : event_type,
+                    EVENT_ATTRIBUTES.SEVERITY : StatusMapper.EVENT_TO_SEVERITY_MAPPING[event_type],
+                    EVENT_ATTRIBUTES.SITE_ID : self.site_id, # TODO: Should be fetched from confstore
+                    EVENT_ATTRIBUTES.RACK_ID : self.rack_id, # TODO: Should be fetched from confstore
+                    EVENT_ATTRIBUTES.CLUSTER_ID : self.cluster_id, # TODO: Should be fetched from confstore
+                    EVENT_ATTRIBUTES.STORAGESET_ID : node_id,
+                    EVENT_ATTRIBUTES.NODE_ID : node_id,
+                    EVENT_ATTRIBUTES.HOST_ID : node_id,
+                    EVENT_ATTRIBUTES.RESOURCE_TYPE : resource_type,
+                    EVENT_ATTRIBUTES.TIMESTAMP : timestamp,
+                    EVENT_ATTRIBUTES.RESOURCE_ID : node_id,
+                    EVENT_ATTRIBUTES.SPECIFIC_INFO : specific_info
+                }
+
+        Log.debug(f"Parsed {event} schema")
+        # Log.error(f"########## Parsed {event} schema")
+        return event
+
+    @staticmethod
+    def _get_sys_health_key_val(key_to_search: str, conf_store) -> (dict,int,str):
+        """
+        Fetches the value from confstore for the given key
+        returns value in dictionary format assoscoiated with
+        the key. and value of pod_restart_val assoscaited with
+        that key(node)
+        conf_store.get will return following strcture
+        {'cortx/ha/v1/cortx/ha/system/cluster/d6b77d52dc0a4fdc8dabd036f57b49eb/site/1/rack/1/node/8f6e76bd96f44fdd8abc7fd7589bf925/health': '{"events": [{"event_timestamp": "1640685608", "created_timestamp": "1640685610", "status": "online", "specific_info": {"generation_id": "cortx-data-pod-ssc-vm-g2-rhev4-2693-68fb57f57-h9kzf", "pod_restart": 0}}], "action": {"modified_timestamp": "1640685610", "status": "pending"}
+        """
+        sys_health_key_val = {}
+        current_pod_restart_val = None
+        sys_health_key_val = conf_store.get(key_to_search)
+        if sys_health_key_val:
+            events_str = sys_health_key_val["cortx/ha/v1/"+key_to_search]
```

Comment:
healthmanager already has functions get_key and set_key . Please use those

Reply:
Yes. Modified the code